internal/provider/common/network/network.go

```diff
@@ -0,0 +1,233 @@
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+
+		"name": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The name of the network. This value must be unique within the `VDC` or `VDC Group` that owns the network.",
+		},
+		"description": schema.StringAttribute{
+			Optional:            true,
+			MarkdownDescription: "A description of the network.",
+		},
+		"gateway": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "(Force replacement) The gateway IP address for the network. This value define also the network IP range with the prefix length.",
+			Validators: []validator.String{
+				fstringvalidator.IsIP(),
+			},
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
+			},
+		},
+		"prefix_length": schema.Int64Attribute{
+			Required:            true,
+			MarkdownDescription: "(Force replacement) The prefix length for the network. This value must be a valid prefix length for the network IP range.(e.g. /24 for netmask 255.255.255.0)",
+			Validators: []validator.Int64{
+				int64validator.Between(1, 32),
+			},
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.RequiresReplace(),
+			},
+		},
+		"dns1": schema.StringAttribute{
+			Optional:            true,
+			MarkdownDescription: "The primary DNS server IP address for the network.",
+			Validators: []validator.String{
+				fstringvalidator.IsIP(),
+			},
+		},
+		"dns2": schema.StringAttribute{
+			Optional:            true,
+			MarkdownDescription: "The secondary DNS server IP address for the network.",
+			Validators: []validator.String{
+				fstringvalidator.IsIP(),
+			},
+		},
+		"dns_suffix": schema.StringAttribute{
+			Optional:            true,
+			MarkdownDescription: "The DNS suffix for the network.",
+		},
+		"static_ip_pool": schema.SetNestedAttribute{
+			Optional:            true,
+			MarkdownDescription: "A set of static IP pools to be used for this network.",
+			NestedObject: schema.NestedAttributeObject{
+				Attributes: map[string]schema.Attribute{
+					"start_address": schema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "The start address of the IP pool. This value must be a valid IP address in the network IP range.",
+						Validators: []validator.String{
+							fstringvalidator.IsIP(),
+						},
+					},
+					"end_address": schema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "The end address of the IP pool. This value must be a valid IP address in the network IP range.",
+						Validators: []validator.String{
+							fstringvalidator.IsIP(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	switch params.typeNetwork {
+	case Routed:
+		// Add routed network specific attributes to the schema
+		_schema.MarkdownDescription = "Provides a CloudAvenue Org VDC routed Network. This can be used to create, modify, and delete routed VDC networks."
```

Comment:
```suggestion
		_schema.MarkdownDescription = "Provides a Cloud Avenue Org VDC routed Network. This can be used to create, modify, and delete routed VDC networks."
```